lib/src/Base/Stat/HMatrixImplementation.cxx

```diff
@@ -313,15 +321,17 @@ HMatrixImplementation::assemble(const HMatrixRealAssemblyFunction& f, char symme
   else if (compressionMethod == "AcaRandom")
     ctx_assemble.compression = hmat_create_compression_aca_random(assemblyEpsilon);
   else
-    LOGWARN( OSS() << "Unknown compression method: " << compressionMethod << ". Valid values are: Svd, AcaFull, AcaPartial or AcaPlus");
+    throw InvalidArgumentException(HERE) << "Unknown compression method: " << compressionMethod << ". Valid values are: Svd, AcaFull, AcaPartial, AcaPlus or AcaRandom";
 
-  static_cast<hmat_interface_t*>(hmatInterface_.get())->assemble_generic(static_cast<hmat_matrix_t*>(hmat_), &ctx_assemble);
+  int rc = static_cast<hmat_interface_t *>(hmatInterface_.get())->assemble_generic(static_cast<hmat_matrix_t *>(hmat_), &ctx_assemble);
+  if (rc != 0)
+    throw InternalException(HERE) << "In HMatrix::assemble, something went wrong";
```

Comment:
HMAT specialists may perhaps be interested in the exact return code, so I propose to include it in the error message. Something like this:
```suggestion
    throw InternalException(HERE) << "In HMatrix::assemble, something went wrong, got return code " << rc;
```

Reply:
looking at the code rc will be either 0 or 1, so not mandatory imo

Reply:
OK, so not very informative. Could it change in the future?

Reply:
I dont know

Reply:
no it is useless. There is only a `rc` and no associated msg